tests/kuttl/common/assert-sample-deployment.yaml

```diff
@@ -205,10 +205,23 @@ status:
   databaseHostname: openstack
   readyCount: 1
 ---
+# the actual addresses of the apiEndpoints are platform specific, so we can't rely on
+# kuttl asserts to check them. This short script gathers the addresses and checks that
+# the three endpoints are defined and their addresses follow the default pattern
 apiVersion: kuttl.dev/v1beta1
 kind: TestAssert
 namespaced: true
 commands:
+  - script: |
+      template='{{.status.apiEndpoints.ironic.public}}{{":"}}{{.status.apiEndpoints.ironic.admin}}{{":"}}{{.status.apiEndpoints.ironic.internal}}{{"\n"}}'
+      regex="http:\/\/ironic-public-openstack\.apps.*:http:\/\/ironic-admin-openstack\.apps.*:http:\/\/ironic-internal-openstack\.apps.*"
+      apiEndpoints=$(oc get -n openstack ironics.ironic.openstack.org ironic -o go-template="$template")
+      matches=$(echo "$apiEndpoints" | sed -e "s?$regex??")
+      if [ -z "$matches" ]; then
+        exit 0
+      else
+        exit 1
+      fi
   - script: |
       set -x
       RETURN_CODE=0
```

Comment:
I'm no understanding why this fails in non crc environments.  I don't see any crc specific things.